### PR TITLE
docs: "Upgrading frontend dependencies with confidence" blog post

### PR DESCRIPTION
**Pull request by slorber**
## Motivation

Blog post to explain how to safely upgrade Docusaurus (or anything else) and catch visual regressions

A workflow we recommend for users upgrading complex sites from Docusaurus v2 to v3.

### Test links

https://deploy-preview-9330--docusaurus-2.netlify.app/blog/upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing/

### CODE REVIEW

**Comment by Josh-Cena** [comment]
LMK when you are ready for editorial reviews

**Comment by slorber** [comment]
> LMK when you are ready for editorial reviews

Thanks!

Feel free to edit the first sections already written. Please do not make it too verbose I prefer it to be concise and to the point, easy to skim ;)

**Comment by slorber** [comment]
Ready to review @Josh-Cena 

I'll post it tomorrow morning. If you find issues after we can still edit it.

File: website/blog/2023/09-22-upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing/index.mdx [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+      # Your job steps here ...
+```
+
+There are many options to explore, such as [triggering the workflow manually](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow) or [only when files matching a specific pattern are modified](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#example-including-paths).
+
+## Conclusion
+
+I believe **visual regression testing is underused** in the frontend ecosystem.
+
+Taking full page screenshots is a **low-hanging fruit** easy to set up, and can help you **catch a new class of bugs** that your regular test suite would miss. This technique shines not only for npm package upgrades, but also for **any kind of refactoring** that shouldn't change the user interface.
```
Comment (Josh-Cena) [inline website/blog/2023/09-22-upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing/index.mdx:377]:
```suggestion
Taking full-page screenshots is a **low-hanging fruit** that is easy to set up and can help you **catch a new class of bugs** that your regular test suite would miss. This technique shines not only for npm package upgrades, but also for **any kind of refactoring** that shouldn't change the user interface.
```

File: website/blog/2023/09-22-upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing/index.mdx [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  }
+  function removeLeadingSlash(str: string): string {
+    return str.startsWith('/') ? str.slice(1) : str;
+  }
+  pathname = removeTrailingSlash(pathname);
+  pathname = removeLeadingSlash(pathname);
+  if (pathname === '') {
+    return 'index';
+  }
+  return pathname;
```
[the remark this answers is not in the document]
Comment (slorber) [reply website/blog/2023/09-22-upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing/index.mdx:225]: Thanks :)

You can submit a PR to our own repo to also simplify this then 😄

**Comment by slorber** [comment]
Thanks for the review 😃 👍

**Comment by Josh-Cena** [comment]
Looks great to me

**Pull request merged**